lw/memory/buffer_view.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <cstdint>
+#include <string_view>
+
+#include "lw/memory/buffer.h"
+
+namespace lw {
+
+class BufferView {
+public:
+  BufferView() = default;
+  BufferView(BufferView&&) = default;
+  BufferView(const BufferView& other): BufferView{other._buffer} {}
+
+  BufferView& operator=(BufferView&&) = default;
+  BufferView& operator=(const BufferView& other) {
+    _buffer = Buffer{
+      const_cast<std::uint8_t*>(other._buffer.data()),
+      other._buffer.size(),
+      /*own_data=*/false
+    };
+    return *this;
+  }
+
+  BufferView(const Buffer& buffer):
+    _buffer{
+      const_cast<std::uint8_t*>(buffer.data()),
+      buffer.size(),
+      /*own_data=*/false
```

Comment:
This is a convention commonly used when passing literal values (in this case `false`) as parameters. When using variables, the names associated with them are self-documenting. For example, it is clear that the previous two arguments are the buffer's data and size respectively. This is clear because they are `buffer.data()` and `buffer.size()`. The argument `false` is meaningless by itself, what am I making false? The `/*x=*/` syntax is making it clear that I am setting the `own_data` parameter to false. For the `Buffer` class, this means the buffer will not claim ownership of the memory I am passing it.